language-support/java/bindings/src/main/java/com/daml/ledger/javaapi/data/SubmitCommandsRequest.java

```diff
@@ -261,6 +265,96 @@ private static CommandsOuterClass.Commands toProto(
     return builder.build();
   }
 
+  private static CommandsOuterClass.Commands toProto(
+      @NonNull String ledgerId,
+      @NonNull Optional<String> submissionId,
+      @NonNull CommandsSubmission commandsSubmission) {
+
+    if (commandsSubmission.getActAs().size() == 0) {
+      throw new IllegalArgumentException("actAs must have at least one element");
+    }
+    ArrayList<CommandsOuterClass.Command> commandsConverted =
+        new ArrayList<>(commandsSubmission.getCommands().size());
```

Comment:
I think this is incorrect because `HasCommands` can have many commands.

Reply:
I previously wrote a `toCommands()` to solve this but `HasCommands` has this too so I've updated the code to use that